File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/FiatCurrencies.kt
package com.example.kotlinmpplib

import breez_liquid_sdk.*

class FiatCurrencies {
    fun listFiatCurrencies(sdk: BindingLiquidSdk) {
        // ANCHOR: list-fiat-currencies
        try {
            val fiatCurrencyList = sdk.listFiatCurrencies()
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-fiat-currencies
    }

    fun fetchFiatRates(sdk: BindingLiquidSdk) {
        // ANCHOR: fetch-fiat-rates
        try {
            val fiatRatesMap = sdk.fetchFiatRates()
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: fetch-fiat-rates
    }
}

File: snippets/wasm/fiat_currencies.ts
import {
  type BindingLiquidSdk
} from '@breeztech/breez-sdk-liquid'

const exampleListCurrencies = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: list-fiat-currencies
  const fiatCurrencies = await sdk.listFiatCurrencies()
  // ANCHOR_END: list-fiat-currencies
}

const exampleFetchRates = async (sdk: BindingLiquidSdk) => {
  // ANCHOR: fetch-fiat-rates
  const fiatRates = await sdk.fetchFiatRates()
  // ANCHOR_END: fetch-fiat-rates
}
